app/models/synthetic_category.rb

```diff
@@ -0,0 +1,22 @@
+class SyntheticCategory < ApplicationRecord
+  belongs_to :contest
+  belongs_to :regular_category, class_name: 'Category'
+  serialize :regular_category_flights
+  serialize :synthetic_category_flights
+
+  def find_or_create
+    last_seq = Category.pluck(:sequence).max
```

Comment:
I assume that the number of Category records will remain small. But if is going to become significant, a more efficient construct is `Category.maximum(:sequence)` which makes the database do all the work.

Reply:
Aha! Thank you! I was thrashing a little for that and settled on something that worked.